core/shared/src/main/scala/matryoshka/Recursive.scala

```diff
@@ -101,13 +115,42 @@ trait Recursive[T] extends Based[T] {
       : A =
     gcata[(B, ?), A](t)(distZygo(f), g)
 
+  def zygoM[A, B, M[_]: Monad]
+    (t: T)
+    (f: AlgebraM[M, Base, B], g: GAlgebraM[(B, ?), M, Base, A])
+    (implicit BT: Traverse[Base])
+      : M[A] = {
+
+    // NB: Probably this is how `gcataM` should look like
+    def gcataM[W[_]: Comonad : Traverse]
```

Comment:
I think this looks reasonable. How about you replace the existing `gcataM` and see if everything still works with this new def?